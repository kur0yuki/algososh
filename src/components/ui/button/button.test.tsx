import renderer from "react-test-renderer";
import {fireEvent, render as reactRenderer, screen} from '@testing-library/react'
import {Button} from "./button";

describe('Button Tests', () => {
    it('Button without text', () => {
        const button = renderer.create(<Button/>)
        let tree = button.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Button with text', () => {
        const button = renderer.create(<Button text={"Some text"}/>)
        let tree = button.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Blocked button', () => {
        const button = renderer.create(<Button text={"Some text"} disabled={true}/>)
        let tree = button.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Button with loader', () => {
        const button = renderer.create(<Button text={"Some text"} isLoader={true}/>)
        let tree = button.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('OnClick Action works', () => {
        const fn = jest.fn(() => 5)
        reactRenderer(<Button text={"Some text"} onClick={fn}/>)

        fireEvent.click(screen.getByText("Some text"))
        expect(fn).toHaveBeenCalledTimes(1)
    })
})