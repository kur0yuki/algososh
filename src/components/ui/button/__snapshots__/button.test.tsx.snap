// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests Blocked button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    Some text
  </p>
</button>
`;

exports[`Button Tests Button with loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button Tests Button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Some text
  </p>
</button>
`;

exports[`Button Tests Button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
